### ISSUE TITLE
`invalid setDrawableSize` on MacOS Sonoma 14.0

### ISSUE THREAD

**Issue by AzurIce** [iced-rs/iced#2113]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

My Iced project not working on MacOS Sonoma 14.0 with the following error:

```
2023-10-12 10:29:32.396 iced-demo[3476:39547] CAMetalLayer ignoring invalid setDrawableSize width=4294967295.000000 height=4294967295.000000
thread 'main' panicked at /Users/azurice/.cargo/git/checkouts/glyphon-bce1d8fb6a1f7cb1/20f0f8f/src/text_render.rs:224:33:
attempt to add with overflow
```

And I tested some examples on branch master:
- some of them get panicked as soon as the windows appear. (tour)
- some of them are initially a blank window, and only after I resize the window it turns normal. (counter)
- Some of them seems normal. (todos)

It seems like a `winit` problem? I guess upgrading `winit` to v0.28.7 should fix it.

See https://github.com/rust-windowing/winit/issues/2876

I pasted a complete log of running example `tour` below.

### What is the expected behavior?

Panic not happens, and everything behaves normal

### Version

master

### Operating System

macOS

### Do you have any log output?

```shell
➜  iced git:(master) ✗ cargo run --package tour
    Finished dev [unoptimized + debuginfo] target(s) in 0.09s
     Running `target/debug/tour`
2023-10-12T02:59:40.223014Z  INFO wgpu_hal::gles::egl: Unable to open libEGL: Library(DlOpen { desc: "dlopen(libEGL.dylib, 0x0005): tried: \'libEGL.dylib\' (no such file), \'/System/Volumes/Preboot/Cryptexes/OSlibEGL.dylib\' (no such file), \'/usr/lib/libEGL.dylib\' (no such file, not in dyld cache), \'libEGL.dylib\' (no such file), \'/Users/azurice/Dev/iced/target/debug/build/objc_exception-eb6ed28b25f2f2fa/out/libEGL.dylib\' (no such file), \'/Users/azurice/Dev/iced/target/debug/deps/libEGL.dylib\' (no such file), \'/Users/azurice/Dev/iced/target/debug/libEGL.dylib\' (no such file), \'/Users/azurice/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libEGL.dylib\' (no such file), \'/Users/azurice/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/libEGL.dylib\' (no such file), \'/Library/Developer/CommandLineTools/usr/lib//libEGL.dylib\' (no such file)" })
2023-10-12T02:59:40.224111Z  INFO iced_wgpu::window::compositor: Settings {
    present_mode: AutoVsync,
    internal_backend: Backends(
        VULKAN | GL | METAL | DX12 | DX11 | BROWSER_WEBGPU,
    ),
    default_font: Font {
        family: SansSerif,
        weight: Normal,
        stretch: Normal,
        style: Normal,
        monospaced: false,
    },
    default_text_size: Pixels(
        16.0,
    ),
    antialiasing: None,
}
2023-10-12T02:59:40.229225Z  INFO wgpu_core::instance: Adapter Metal AdapterInfo { name: "Apple M2 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
2023-10-12T02:59:40.231728Z  INFO iced_wgpu::window::compositor: Available adapters: [
    AdapterInfo {
        name: "Apple M2 Max",
        vendor: 0,
        device: 0,
        device_type: IntegratedGpu,
        driver: "",
        driver_info: "",
        backend: Metal,
    },
]
2023-10-12T02:59:40.232673Z  INFO wgpu_core::instance: Adapter Metal AdapterInfo { name: "Apple M2 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
2023-10-12T02:59:40.232748Z  INFO iced_wgpu::window::compositor: Selected: AdapterInfo {
    name: "Apple M2 Max",
    vendor: 0,
    device: 0,
    device_type: IntegratedGpu,
    driver: "",
    driver_info: "",
    backend: Metal,
}
2023-10-12T02:59:40.233436Z  INFO iced_wgpu::window::compositor: Selected format: Bgra8UnormSrgb
2023-10-12T02:59:40.330102Z  INFO cosmic_text::font::system::std: Parsed 1431 font faces in 93ms.
2023-10-12T02:59:40.337969Z  INFO wgpu_core::device::global: Created buffer Valid((0, 1, Metal)) with BufferDescriptor { label: Some("glyphon params"), size: 16, usage: BufferUsages(COPY_DST | UNIFORM), mapped_at_creation: false }
2023-10-12T02:59:40.338663Z  INFO wgpu_core::device::global: Created texture Valid((0, 1, Metal)) with TextureDescriptor { label: Some("glyphon atlas"), size: Extent3d { width: 256, height: 256, depth_or_array_layers: 1 }, mip_level_count: 1, sample_count: 1, dimension: D2, format: Rgba8UnormSrgb, usage: TextureUsages(COPY_DST | TEXTURE_BINDING), view_formats: [] }
2023-10-12T02:59:40.339203Z  INFO wgpu_core::device::global: Created texture Valid((1, 1, Metal)) with TextureDescriptor { label: Some("glyphon atlas"), size: Extent3d { width: 256, height: 256, depth_or_array_layers: 1 }, mip_level_count: 1, sample_count: 1, dimension: D2, format: R8Unorm, usage: TextureUsages(COPY_DST | TEXTURE_BINDING), view_formats: [] }
2023-10-12T02:59:40.339835Z  INFO wgpu_core::device::global: Created buffer Valid((1, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu::quad vertex buffer"), size: 32, usage: BufferUsages(VERTEX), mapped_at_creation: true }
2023-10-12T02:59:40.340357Z  INFO wgpu_core::device::global: Created buffer Valid((2, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu::quad index buffer"), size: 12, usage: BufferUsages(INDEX), mapped_at_creation: true }
2023-10-12T02:59:40.345030Z  INFO wgpu_core::device::global: Created render pipeline Valid((0, 1, Metal)) with RenderPipelineDescriptor { label: Some("iced_wgpu.quad.solid.pipeline"), layout: Some((1, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (1, 1, Metal), entry_point: "solid_vs_main" }, buffers: [VertexBufferLayout { array_stride: 8, step_mode: Vertex, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 0 }] }, VertexBufferLayout { array_stride: 68, step_mode: Instance, attributes: [VertexAttribute { format: Float32x4, offset: 0, shader_location: 1 }, VertexAttribute { format: Float32x2, offset: 16, shader_location: 2 }, VertexAttribute { format: Float32x2, offset: 24, shader_location: 3 }, VertexAttribute { format: Float32x4, offset: 32, shader_location: 4 }, VertexAttribute { format: Float32x4, offset: 48, shader_location: 5 }, VertexAttribute { format: Float32, offset: 64, shader_location: 6 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Cw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (1, 1, Metal), entry_point: "solid_fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
2023-10-12T02:59:40.352603Z  INFO wgpu_core::device::global: Created render pipeline Valid((1, 1, Metal)) with RenderPipelineDescriptor { label: Some("iced_wgpu.quad.gradient.pipeline"), layout: Some((2, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (1, 2, Metal), entry_point: "gradient_vs_main" }, buffers: [VertexBufferLayout { array_stride: 8, step_mode: Vertex, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 0 }] }, VertexBufferLayout { array_stride: 148, step_mode: Instance, attributes: [VertexAttribute { format: Uint32x4, offset: 0, shader_location: 1 }, VertexAttribute { format: Uint32x4, offset: 16, shader_location: 2 }, VertexAttribute { format: Uint32x4, offset: 32, shader_location: 3 }, VertexAttribute { format: Uint32x4, offset: 48, shader_location: 4 }, VertexAttribute { format: Uint32x4, offset: 64, shader_location: 5 }, VertexAttribute { format: Float32x4, offset: 80, shader_location: 6 }, VertexAttribute { format: Float32x4, offset: 96, shader_location: 7 }, VertexAttribute { format: Float32x4, offset: 112, shader_location: 8 }, VertexAttribute { format: Float32x4, offset: 128, shader_location: 9 }, VertexAttribute { format: Float32, offset: 144, shader_location: 10 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Cw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (1, 2, Metal), entry_point: "gradient_fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
2023-10-12T02:59:40.353470Z  INFO wgpu_core::device::global: Created render pipeline Valid((2, 1, Metal)) with RenderPipelineDescriptor { label: Some("iced_wgpu::triangle::solid pipeline"), layout: Some((3, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (1, 3, Metal), entry_point: "solid_vs_main" }, buffers: [VertexBufferLayout { array_stride: 24, step_mode: Vertex, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 0 }, VertexAttribute { format: Float32x4, offset: 8, shader_location: 1 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Cw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (1, 3, Metal), entry_point: "solid_fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
2023-10-12T02:59:40.357917Z  INFO wgpu_core::device::global: Created render pipeline Valid((3, 1, Metal)) with RenderPipelineDescriptor { label: Some("iced_wgpu.triangle.gradient.pipeline"), layout: Some((4, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (1, 4, Metal), entry_point: "gradient_vs_main" }, buffers: [VertexBufferLayout { array_stride: 104, step_mode: Vertex, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 0 }, VertexAttribute { format: Uint32x4, offset: 8, shader_location: 1 }, VertexAttribute { format: Uint32x4, offset: 24, shader_location: 2 }, VertexAttribute { format: Uint32x4, offset: 40, shader_location: 3 }, VertexAttribute { format: Uint32x4, offset: 56, shader_location: 4 }, VertexAttribute { format: Uint32x4, offset: 72, shader_location: 5 }, VertexAttribute { format: Float32x4, offset: 88, shader_location: 6 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Cw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (1, 4, Metal), entry_point: "gradient_fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
2023-10-12T02:59:40.359182Z  INFO wgpu_core::device::global: Created render pipeline Valid((4, 1, Metal)) with RenderPipelineDescriptor { label: Some("iced_wgpu::image pipeline"), layout: Some((5, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (1, 5, Metal), entry_point: "vs_main" }, buffers: [VertexBufferLayout { array_stride: 8, step_mode: Vertex, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 0 }] }, VertexBufferLayout { array_stride: 36, step_mode: Instance, attributes: [VertexAttribute { format: Float32x2, offset: 0, shader_location: 1 }, VertexAttribute { format: Float32x2, offset: 8, shader_location: 2 }, VertexAttribute { format: Float32x2, offset: 16, shader_location: 3 }, VertexAttribute { format: Float32x2, offset: 24, shader_location: 4 }, VertexAttribute { format: Sint32, offset: 32, shader_location: 5 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Cw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (1, 5, Metal), entry_point: "fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
2023-10-12T02:59:40.359235Z  INFO wgpu_core::device::global: Created buffer Valid((3, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu::image vertex buffer"), size: 32, usage: BufferUsages(VERTEX), mapped_at_creation: true }
2023-10-12T02:59:40.359270Z  INFO wgpu_core::device::global: Created buffer Valid((4, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu::image index buffer"), size: 12, usage: BufferUsages(INDEX), mapped_at_creation: true }
2023-10-12T02:59:40.359317Z  INFO wgpu_core::device::global: Created texture Valid((2, 1, Metal)) with TextureDescriptor { label: Some("iced_wgpu::image texture atlas"), size: Extent3d { width: 2048, height: 2048, depth_or_array_layers: 1 }, mip_level_count: 1, sample_count: 1, dimension: D2, format: Rgba8UnormSrgb, usage: TextureUsages(COPY_SRC | COPY_DST | TEXTURE_BINDING), view_formats: [] }
2023-10-12T02:59:40.372928Z  INFO wgpu_core::device::global: configuring surface with SurfaceConfiguration { usage: TextureUsages(RENDER_ATTACHMENT), format: Bgra8UnormSrgb, width: 2048, height: 1536, present_mode: AutoVsync, alpha_mode: Auto, view_formats: [] }
2023-10-12T02:59:40.372966Z  INFO wgpu_core::device::global: Automatically choosing presentation mode by rule AutoVsync. Chose Fifo
2023-10-12T02:59:40.372981Z  INFO wgpu_core::device::global: Automatically choosing alpha mode by rule Auto. Chose Opaque
2023-10-12T02:59:40.372987Z  INFO wgpu_hal::metal::surface: build swapchain SurfaceConfiguration { swap_chain_size: 3, present_mode: Fifo, composite_alpha_mode: Opaque, format: Bgra8UnormSrgb, extent: Extent3d { width: 2048, height: 1536, depth_or_array_layers: 1 }, usage: TextureUses(COLOR_TARGET), view_formats: [] }
2023-10-12T02:59:40.399901Z  INFO wgpu_core::device::global: configuring surface with SurfaceConfiguration { usage: TextureUsages(RENDER_ATTACHMENT), format: Bgra8UnormSrgb, width: 4294967295, height: 4294967295, present_mode: AutoVsync, alpha_mode: Auto, view_formats: [] }
2023-10-12T02:59:40.399943Z  WARN wgpu_core::device::global: Requested size 4294967295x4294967295 is outside of the supported range: Extent3d { width: 4, height: 4, depth_or_array_layers: 1 }..=Extent3d { width: 16384, height: 16384, depth_or_array_layers: 1 }
2023-10-12T02:59:40.399954Z  INFO wgpu_core::device::global: Automatically choosing presentation mode by rule AutoVsync. Chose Fifo
2023-10-12T02:59:40.399960Z  INFO wgpu_core::device::global: Automatically choosing alpha mode by rule Auto. Chose Opaque
2023-10-12T02:59:40.399966Z  INFO wgpu_hal::metal::surface: build swapchain SurfaceConfiguration { swap_chain_size: 3, present_mode: Fifo, composite_alpha_mode: Opaque, format: Bgra8UnormSrgb, extent: Extent3d { width: 4294967295, height: 4294967295, depth_or_array_layers: 1 }, usage: TextureUses(COLOR_TARGET), view_formats: [] }
2023-10-12 10:59:40.399 tour[4938:68323] CAMetalLayer ignoring invalid setDrawableSize width=4294967295.000000 height=4294967295.000000
2023-10-12T02:59:40.400978Z  INFO wgpu_core::device::global: Created buffer Valid((5, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu::quad uniforms buffer"), size: 80, usage: BufferUsages(COPY_DST | UNIFORM), mapped_at_creation: false }
2023-10-12T02:59:40.401273Z  INFO wgpu_core::device::global: Created buffer Valid((6, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu.quad.solid.buffer"), size: 262144, usage: BufferUsages(COPY_DST | VERTEX), mapped_at_creation: false }
2023-10-12T02:59:40.401295Z  INFO wgpu_core::device::global: Created buffer Valid((7, 1, Metal)) with BufferDescriptor { label: Some("iced_wgpu.quad.gradient.buffer"), size: 524288, usage: BufferUsages(COPY_DST | VERTEX), mapped_at_creation: false }
2023-10-12T02:59:40.401393Z  INFO wgpu_core::device::global: Created buffer Valid((8, 1, Metal)) with BufferDescriptor { label: Some("glyphon vertices"), size: 4096, usage: BufferUsages(COPY_DST | VERTEX), mapped_at_creation: false }
2023-10-12T02:59:40.401416Z  INFO wgpu_core::device::global: Created buffer Valid((9, 1, Metal)) with BufferDescriptor { label: Some("glyphon indices"), size: 4096, usage: BufferUsages(COPY_DST | INDEX), mapped_at_creation: false }
2023-10-12T02:59:40.402402Z  INFO wgpu_core::device::global: Created render pipeline Valid((5, 1, Metal)) with RenderPipelineDescriptor { label: Some("glyphon pipeline"), layout: Some((0, 1, Metal)), vertex: VertexState { stage: ProgrammableStageDescriptor { module: (0, 1, Metal), entry_point: "vs_main" }, buffers: [VertexBufferLayout { array_stride: 28, step_mode: Vertex, attributes: [VertexAttribute { format: Sint32x2, offset: 0, shader_location: 0 }, VertexAttribute { format: Uint32, offset: 8, shader_location: 1 }, VertexAttribute { format: Uint32, offset: 12, shader_location: 2 }, VertexAttribute { format: Uint32, offset: 16, shader_location: 3 }, VertexAttribute { format: Uint32, offset: 20, shader_location: 4 }, VertexAttribute { format: Float32, offset: 24, shader_location: 5 }] }] }, primitive: PrimitiveState { topology: TriangleList, strip_index_format: None, front_face: Ccw, cull_mode: None, unclipped_depth: false, polygon_mode: Fill, conservative: false }, depth_stencil: None, multisample: MultisampleState { count: 1, mask: 18446744073709551615, alpha_to_coverage_enabled: false }, fragment: Some(FragmentState { stage: ProgrammableStageDescriptor { module: (0, 1, Metal), entry_point: "fs_main" }, targets: [Some(ColorTargetState { format: Bgra8UnormSrgb, blend: Some(BlendState { color: BlendComponent { src_factor: SrcAlpha, dst_factor: OneMinusSrcAlpha, operation: Add }, alpha: BlendComponent { src_factor: One, dst_factor: OneMinusSrcAlpha, operation: Add } }), write_mask: ColorWrites(RED | GREEN | BLUE | ALPHA) })] }), multiview: None }
thread 'main' panicked at /Users/azurice/.cargo/git/checkouts/glyphon-bce1d8fb6a1f7cb1/20f0f8f/src/text_render.rs:224:33:
attempt to add with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
2023-10-12T02:59:40.412761Z  INFO wgpu_core::global: Dropping Global
2023-10-12T02:59:40.413049Z  INFO wgpu_core::device: Destroying 0 command encoders
```

**Comment by AzurIce** [comment]
After some trying and diving, I think we may need to do these works in order to solve the problem:
- make the upstream of [`glyphon`](https://github.com/grovesNL/glyphon/tree/main) to pump its `winit` version to `v0.28.7`
- merge the branch `v0.28.x` from the upstream of `winit` to our iced forked one
- update `Cargo.toml`

I'm new to participating an open source project, so I'm not sure whether I'm following a correct way of contribution, so please forgive me if I made some mistake.

**Comment by AzurIce** [comment]
I've opened a pull request on [`glyphon`](https://github.com/grovesNL/glyphon) to pump the `winit` version:  https://github.com/grovesNL/glyphon/pull/57

**Comment by oneElectron** [comment]
I have a different issue, but I think my issue might be related.
If I run some of the examples in the examples directory (I have tested game_of_life and loading_spinners) I get a blank window.
This issue seemed to be related so I back ported this fix: [https://github.com/rust-windowing/winit/commit/55f60a35b126f09b31e74a0b5dee99fcadeef2a6](https://github.com/rust-windowing/winit/commit/55f60a35b126f09b31e74a0b5dee99fcadeef2a6) and that solved it.

I am not 100% percent certain the two issues are related and I have never contributed to iced, so some direction would be nice

**Comment by AzurIce** [comment]
> I have a different issue, but I think my issue might be related. If I run some of the examples in the examples directory (I have tested game_of_life and loading_spinners) I get a blank window. 

This problem also appears when I run the `counter` example.

> I am not 100% percent certain the two issues are related and I have never contributed to iced, so some direction would be nice

So, I believe these two issues are caused by the same bug from `winit`, and the commit you mentioned should fix it.

I forked the repository of iced and the iced-forked winit, and merged the branch `v0.28.x` from [rust-windowing/winit](https://github.com/rust-windowing/winit) (which has https://github.com/iced-rs/winit/commit/55f60a35b126f09b31e74a0b5dee99fcadeef2a6 and https://github.com/iced-rs/winit/commit/2ebbfab6a47f93f76841553dcb3cc53683e4955c, the latter just something about change logs and versioning), then I modified `Cargo.toml` of iced, and it solves the problem. I'm now also using this way in my project.

However, I'm not sure how to contribute these changes to iced (afraid of messing up some thing), so , yes, some direction would be nice.

Maybe @hecrj can have a look?
